corehq/apps/dump_reload/sql/dump.py

```diff
@@ -237,9 +251,10 @@ def dump(self, output_stream):
             stats_counter=stats,
             stdout=self.stdout,
         )
+
         JsonLinesSerializer().serialize(
             objects,
-            use_natural_foreign_keys=False,
+            use_natural_foreign_keys=True,
```

Comment:
Yeah good question. I was actually going to ask @snopoke if he had any ideas but it was also introduced a long time ago so wasn't sure how much context he would have. Looks like it was introduced [here](https://github.com/dimagi/commcare-hq/commit/506d966048f066de259c6e985a0f744ecdf9183e) if that helps jog anyone's memory.

The reason I didn't mention it is because my understanding is that this is a pretty safe setting. It _only_ impacts models that have `natural_key` defined, **and** are part of foreign key relationships that use the primary key. Our form and case models are the only place where both `natural_key` is defined and foreign key relationships are used. All relationships rely on form_id and case_id which are not the primary key for XFormInstance and CommCareCase respectively.

Even if they did rely on the primary key, we would want this setting enabled for the same reason we enable `use_natural_primary_key`, so that the natural key would be used to identify the object in the foreign key relationship. We would have to also define `get_by_natural_key` though to ensure the models are loaded properly. I can explain more about that method in my response to your other question.

Reply:
Though looking at [django source code](https://github.com/django/django/blob/stable/3.2.x/django/core/serializers/python.py#L51-L60), I'm a bit confused about why the foreign keys for case and form objects aren't impacted by this. The code seems to suggest if `use_natural_foreign_keys` is defined and a `natural_key` method is defined, use the natural key instead of the value set for that field.

I think this is actually what is happening, and the "problem" is that our `natural_key` methods return a string (in the form of the case_id or form_id), as opposed to a tuple as the [docs specify](https://docs.djangoproject.com/en/3.2/topics/serialization/#serialization-of-natural-keys).